pkg/cmd/server/start/node_args.go

```diff
@@ -57,13 +57,19 @@ type NodeArgs struct {
 
 	// Bootstrap is true if the node should rely on the server to set initial configuration.
 	Bootstrap bool
+	// BootstrapConfigName is the name of a config map to read node-config.yaml from.
+	BootstrapConfigName string
+	// BootstrapConfigNamespace is the namespace the config map for bootstrap config is expected to load from.
+	BootstrapConfigNamespace string
 
 	MasterCertDir string
 	ConfigDir     flag.StringFlag
 
 	AllowDisabledDocker bool
 	// VolumeDir is the volume storage directory.
 	VolumeDir string
+	// VolumeDirProvided is set to true if the user has specified this flag.
+	VolumeDirProvided bool
```

Comment:
blech

Reply:
Hold your nose a month or so longer and most of this code dies and goes away and we're all happy.